src/ethereum/paris/spec.py

```diff
@@ -0,0 +1,848 @@
+
+    Accounts that are marked for deletion are processed and destroyed after
+    execution.
+
+    Parameters
+    ----------
+    env :
+        Environment for the Ethereum Virtual Machine.
+    tx :
+        Transaction to execute.
+
+    Returns
+    -------
+    gas_left : `ethereum.base_types.U256`
+        Remaining gas after execution.
+    logs : `Tuple[ethereum.eth_types.Log, ...]`
+        Logs generated during execution.
+    """
+    ensure(validate_transaction(tx), InvalidBlock)
+
+    sender = env.origin
+    sender_account = get_account(env.state, sender)
+
+    if isinstance(tx, FeeMarketTransaction):
+        gas_fee = tx.gas * tx.max_fee_per_gas
+    else:
+        gas_fee = tx.gas * tx.gas_price
+
+    ensure(sender_account.nonce == tx.nonce, InvalidBlock)
+    ensure(sender_account.balance >= gas_fee + tx.value, InvalidBlock)
+    ensure(sender_account.code == bytearray(), InvalidBlock)
+
+    effective_gas_fee = tx.gas * env.gas_price
+
+    gas = tx.gas - calculate_intrinsic_cost(tx)
+    increment_nonce(env.state, sender)
+
+    sender_balance_after_gas_fee = sender_account.balance - effective_gas_fee
+    set_account_balance(env.state, sender, sender_balance_after_gas_fee)
+
+    preaccessed_addresses = set()
+    preaccessed_storage_keys = set()
+    if isinstance(tx, (AccessListTransaction, FeeMarketTransaction)):
+        for (address, keys) in tx.access_list:
+            preaccessed_addresses.add(address)
+            for key in keys:
+                preaccessed_storage_keys.add((address, key))
+
+    message = prepare_message(
+        sender,
+        tx.to,
+        tx.value,
+        tx.data,
+        gas,
+        env,
+        preaccessed_addresses=frozenset(preaccessed_addresses),
+        preaccessed_storage_keys=frozenset(preaccessed_storage_keys),
+    )
+
+    output = process_message_call(message, env)
+
+    gas_used = tx.gas - output.gas_left
+    gas_refund = min(gas_used // 5, output.refund_counter)
+    gas_refund_amount = (output.gas_left + gas_refund) * env.gas_price
+
+    # For non-1559 transactions env.gas_price == tx.gas_price
+    priority_fee_per_gas = env.gas_price - env.base_fee_per_gas
+    transaction_fee = (
+        tx.gas - output.gas_left - gas_refund
+    ) * priority_fee_per_gas
+
+    total_gas_used = gas_used - gas_refund
+
+    # refund gas
+    sender_balance_after_refund = (
+        get_account(env.state, sender).balance + gas_refund_amount
+    )
+    set_account_balance(env.state, sender, sender_balance_after_refund)
+
+    # transfer miner fees
+    coinbase_balance_after_mining_fee = (
+        get_account(env.state, env.coinbase).balance + transaction_fee
+    )
+    if coinbase_balance_after_mining_fee != 0:
+        set_account_balance(
+            env.state, env.coinbase, coinbase_balance_after_mining_fee
+        )
+    elif is_account_empty(env.state, env.coinbase):
+        destroy_account(env.state, env.coinbase)
```

Comment:
It is not obvious to me why we're changing this behaviour, especially when EIP-3675 says:

> Note: Transaction fee mechanics affecting the block’s beneficiary account MUST remain unchanged.

Reply:
Should we open a bug for this then?

Reply:
I've backported the change all the way to the introduction of account clearing in `spurious_dragon`.